app/Commands/NewCommand.php

```diff
@@ -0,0 +1,511 @@
+use Tivie\OS\Detector;
+
+class NewCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'new
+                        {name : Name of the Laravel project}
+                        {--a|auth : Run make:auth}
+                        {--b|browser : Browser you want to open the project in}
+                        {--c|createdb= : Create a database; pass in sqlite or mysql}
+                        {--d|dev : Choose the dev branch instead of master}
+                        {--e|editor= : Text editor to open the project in}
+                        {--l|link : Create a Valet link to the project directory}
+                        {--m|message= : Set the first commit message}
+                        {--y|node : Set to execute yarn or npm install}
+                        {--p|path= : Base path for the installation, otherwise CWD is used}';
+
+    protected $projectname = '';
+
+    protected $projectpath = '';
+
+    protected $projecturl = '';
+
+    protected $cwd = '';
+
+    protected $basepath = '';
+
+    protected $commitmessage = 'Initial commit.';
+
+    protected $editors_terminal = ['vim', 'vi', 'nano', 'pico', 'ed', 'emacs', 'nvim'];
+
+    protected $editors_gui = ['pstorm', 'subl', 'atom', 'textmate', 'geany'];
+
+    protected $editor = '';
+
+    protected $tools = [];
+
+    protected $dbtypes = ['sqlite', 'mysql'];
+
+    protected $os = null;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Creates a new Laravel application';
+
+    /**
+     * Create a new command instance.
+     */
+    public function __construct(ExecutableFinder $finder)
+    {
+        parent::__construct();
+
+        $this->cwd = getcwd();
+        $this->basepath = $this->cwd;
+
+        $this->os = new Detector;
+        $this->finder = $finder;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle(): void
+    {
+        $this->setup();
+
+        if (! $this->hasTool('laravel')) {
+            $this->error('Unable to find laravel installer so I must exit. One day I will use composer here instead of exiting.');
+            exit;
+        }
+
+        if (is_dir($this->projectpath)) {
+            if (! $this->askToAndRemoveProject()) {
+                $this->error('Goodbye!');
+                exit;
+            }
+        }
+
+        if ($this->option('dev')) {
+            $this->warn('The laravel installation will use the latest developmental branch by passing in --dev');
+            $branch = ' --dev';
+        } else {
+            $branch = '';
+        }
+
+        $command = "laravel new {$this->projectname}$branch";
+
+        $this->info("Creating a new project named {$this->projectname}");
+        $this->info("Executing command '$command' in directory {$this->basepath}");
```

Comment:
Let's do braces around all of our inline variables just for clarity.

Reply:
Done!